### DISCUSSION TITLE
Collection import via cli

### DISCUSSION THREAD

**Discussion by jonocodes** [usebruno/bruno#1139] [Ideas]
My ultimate goal is to maintain some synchronization to my openapi spec. If there is no way to provide a persistent URL for which the spec is regularly checked, I propose a more manual approach.

In the cli I would like to import the spec via a file or URL. That way I could bash script my own automation around how I handle changes to the spec into my collection.

**Comment by n00dleee** [comment]
same feeling here: 

it could be a fantastic if we had a way to manage/update/synchronize collections via some scripts to be triggered either manually or automatically 🔥 

at the moment, it's a blocker for me to migrate all Postman stuff in Bruno because I'm unsure on how to not have to manually replace collections after each openAPI files updates 🥲

**Reply by helloanoop** [reply]
@nicolharico We'll support a command line utility to do this very soon.

I am thinking of something like `bru import --type=openapiv3 service.yaml`
This command would parse the spec and overwrite any new changes.

Also, how are you currently doing this in postman, does postman allow to update collections with openapi file changes ?

**Reply by n00dleee** [reply]
"We'll support a command line utility to do this very soon."
how cool is that 🔥 

We have written some scripts, which are triggered in documentation CI, which use Postman http API to do all the required updates based on openAPI files.

**Reply by veceraj** [reply]
Hey, I am looking forward for this feature! Sounds cool. I'd love to implement something to automatically regenerate .bru files based on openapi.json which I am generating using a composer script. It would be great to later run the Bruno api tests! I could imagine having it as part of some pipeline on PR's? Thanks for all your amazing work 🔥

**Reply by mishioo** [reply]
@helloanoop, what's the status of this feature? Is there any update on it? I'd love to see it. We're looking for a way to automate updates to our collection in Bruno.

**Comment by sushmithasv** [comment]
What is an alternative for checking a schema with tv4?

**Comment by lcfd** [comment]
Hi @helloanoop 👋

If I have to imagine this feature it would be like that:
- A setting in the collection called `OpenAPI URL`—or something similar
- In the context menu of the collection list in the right sidebar
  - Add a menu item called `Refresh using URL`
- When clicked on the item a confirmation popup should spawn
- If confirmed the collection will be recreated using the URL

**Reply by Boxx1483** [reply]
Any news on this feature?

**Comment by RHeynsZa** [comment]
Please give this so that we can generate bruno docs off of generated Swagger/OpenAPI docs

**Comment by JSSAggie** [comment]
I am looking at exposing a public GitHub repo bruno collection from my companies public open api spec.  I would like to use the CLI to do the heavy lifting and then we would write a script to add some of the additional details that we want to support for a better developer experience.  How would you suggest we do that?

I can import the initial one into bruno and have it generate a new collection but how should I handle updates?  I was looking for the CLI to help me out here.

**Comment by rreyn-bruno** [comment]
Hi All - we've just released a set of converters that allows you to programmatically convert a number of items into Bruno collections. [One of those items is an OpenAPI spec. ](https://docs.usebruno.com/converters/overview#openapitobrunoopenapispec-object). 

Would be great to hear feedback on how you're using this once you have it implemented.

**Reply by tseever** [reply]
The converter appears to work nicely, but is there an existing way to export it into the same flat format as the main app does rather than the JSON collection?
